test/mpi/include/mpitest.h

```diff
@@ -58,6 +58,148 @@ typedef struct _MTestDatatype {
     int (*CheckBuf) (struct _MTestDatatype *);
 } MTestDatatype;
 
+typedef struct {
+    const char *typename;
+    MPI_Datatype type;
+} MTestType_t;
+
+#define MTEST_DEFINE_AND_INIT_TYPELIST                              \
```

Comment:
I would prefer avoiding macros if there is an alternative. This case seems to me that using extern variable for `typelist` and/or use a function for `MTEST_INIT_BASIC_SIGNATURE` would be more natural.

Reply:
Ok I will fix that

Reply:
@hajime I have fixed this. Unfortunately I don't have time to extend the interface now. If you want we can cherry pick just this one commit and leave the interface extension to some other time.